### PR TITLE
nmcli: Typecast parameter values to string as required

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

Fixes: #472

Signed-off-by: Marcin Sucharski <marcin.sucharski.github@gmail.com>
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
changelogs/fragments/58115_nmcli.yml
plugins/modules/net_tools/nmcli.py
tests/unit/plugins/modules/net_tools/test_nmcli.py

### CODE REVIEW

**Comment by Akasurde** [comment]
cc @cppmonkey @polatsinan @marcin-sucharski @numas @kbrowder @martingolas @learnitall Could you please review this asap so that I can merge it and backport it for Stable-2.9 release? Thanks in advance.

**Comment by Hilzu** [comment]
Creating an ethernet connection using nmcli module from this PR seems to fail.

```yaml
- name: Create ethernet connection
  nmcli:
    conn_name: Ethernet connection 1
    type: ethernet
    state: present
    ifname: enp0s5
```

```
fatal: [ubuntu-bionic]: FAILED! => {"changed": false, "msg": "Error: failed to modify 802-3-ethernet.mtu: 'None' is out of range [0, 4294967295].\n", "name": "Ethernet connection 1", "rc": 2}
```

**Comment by Akasurde** [comment]
@Hilzu Thanks for testing the PR. I made some changes.

**Comment by learnitall** [comment]
Thanks for putting this together so fast! 

I tried running the playbook snippets I mentioned in #472 using this PR and no `TypeError` was raised. For clarity, here are those snippets again:

```yaml
- name: Add a br-mgmt interface
  nmcli:
    state: present
    conn_name: br-mgmt
    ifname: br-mgmt
    type: bridge
    autoconnect: yes
    ip4: ''
    gw4: ''
    ip6: ''
    gw6: ''
    stp: yes
    priority: 12
    slavepriority: 32
    forwarddelay: 15
    hellotime: 2
    maxage: 20
    ageingtime: 300
    hairpin: yes
    path_cost: 100
```

```yaml
- name: Add a br-mgmt interface
  nmcli:
    state: present
    conn_name: br-mgmt
    ifname: br-mgmt
    type: bridge
    autoconnect: yes
```

And here's my verbatim output with `-vvv`:

```
changed: [my.censored.hostname] => {
    "Exists": "Connections do exist so we are modifying them", 
    "changed": true, 
    "conn_name": "br-mgmt", 
    "invocation": {
        "module_args": {
            "ageingtime": 300, 
            "arp_interval": null, 
            "arp_ip_target": null, 
            "autoconnect": true, 
            "conn_name": "br-mgmt", 
            "dhcp_client_id": null, 
            "dns4": null, 
            "dns4_search": null, 
            "dns6": null, 
            "dns6_search": null, 
            "downdelay": null, 
            "egress": null, 
            "flags": null, 
            "forwarddelay": 15, 
            "gw4": "", 
            "gw6": "", 
            "hairpin": true, 
            "hellotime": 2, 
            "ifname": "br-mgmt", 
            "ingress": null, 
            "ip4": "", 
            "ip6": "", 
            "ip_tunnel_dev": null, 
            "ip_tunnel_local": null, 
            "ip_tunnel_remote": null, 
            "mac": null, 
            "master": null, 
            "maxage": 20, 
            "miimon": null, 
            "mode": "balance-rr", 
            "mtu": null, 
            "path_cost": 100, 
            "primary": null, 
            "priority": 12, 
            "slavepriority": 32, 
            "state": "present", 
            "stp": true, 
            "type": "bridge", 
            "updelay": null, 
            "vlandev": null, 
            "vlanid": null, 
            "vxlan_id": null, 
            "vxlan_local": null, 
            "vxlan_remote": null
        }
    }, 
    "state": "present"
}
```

File: changelogs/fragments/58115_nmcli.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+- nmcli - Typecast parameters to string as required (https://github.com/ansible/ansible/issues/59095).
```
Comment (Andersson007) [inline changelogs/fragments/58115_nmcli.yml:2]:
```suggestion
- nmcli - typecast parameters to string as required (https://github.com/ansible/ansible/issues/59095).
```

**Pull request merged**